import-export-cli/impl/init.go

```diff
@@ -0,0 +1,262 @@
+/*
+*  Copyright (c) WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+ */
+
+package impl
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/Jeffail/gabs"
+	"github.com/go-openapi/loads"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/wso2/product-apim-tooling/import-export-cli/box"
+	v2 "github.com/wso2/product-apim-tooling/import-export-cli/specs/v2"
+	"github.com/wso2/product-apim-tooling/import-export-cli/utils"
+	"gopkg.in/yaml.v2"
+	yaml2 "gopkg.in/yaml.v2"
+)
+
+// Directories to be created during init
+var dirs = []string{
+	utils.InitProjectDefinitions,
+	utils.InitProjectImage,
+	utils.InitProjectDocs,
+	utils.InitProjectSequences,
+	utils.InitProjectSequencesFault,
+	utils.InitProjectSequencesIn,
+	utils.InitProjectSequencesOut,
+	utils.InitProjectClientCertificates,
+	utils.InitProjectClientCertificates,
+	utils.InitProjectInterceptors,
+	utils.InitProjectLibs,
+}
+
+// InitAPIProject function is used to initlialize an API Project
+func InitAPIProject(initCmdOutputDir, initCmdInitialState, initCmdSwaggerPath, initCmdApiDefinitionPath string, isAWSAPI bool) error {
```

Comment:
Shall we avoid adding AWS API specific logic in this function? ideally the `aws init` logic should call this function to create the basic project structure and then add the custom data into this project

Reply:
As per the discussion added a property to validate whether an API is advertised only one before writing the deployment environments via https://github.com/wso2/product-apim-tooling/pull/663/commits/96647b84aefbe91d7ad17c8fe6f3d0aadc74ff04